specification/saas/resource-manager/Microsoft.SaaS/preview/2018-03-01-beta/saas.json

```diff
@@ -0,0 +1,866 @@
+      "description": "Operation List Result Response",
+      "properties": {
+        "value": {
+          "description": "List of SaaS operations supported by the SaaS resource provider.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          }
+        }
+      }
+    },
+    "Operation": {
+      "description": "SaaS REST API operation definition.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Operation name",
+          "type": "string"
+        },
+        "display": {
+          "description": "Display metadata associated with the operation.",
+          "properties": {
+            "provider": {
+              "description": "Service provider: Microsoft SaaS.",
+              "type": "string"
+            },
+            "resource": {
+              "description": "Resource on which the operation is performed.",
+              "type": "string"
+            },
+            "operation": {
+              "description": "Type of the operation: get, read, delete, etc.",
+              "type": "string"
+            },
+            "description": {
+              "description": "Description of the operation.",
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "SaasResource": {
+      "description": "SaaS REST API resource definition.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "description": "saas properties",
+          "type": "object",
+          "allOf": [
+            {
+              "$ref": "#/definitions/SaasProperties"
+            },
+            {
+              "$ref": "#/definitions/SaasCreationProperties"
+            }
+          ]
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the resource"
+        },
+        "id": {
+          "description": "The resource uri",
+          "type": "string"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        },
+        "tags": {
+          "description": "the resource tags.",
+          "$ref": "#/definitions/Tags"
+        }
+      }
+    },
+    "SaasResourceCreation": {
+      "description": "SaaS REST API resource definition for creation.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The resource uri",
+          "type": "string"
+        },
+        "name": {
+          "description": "The resource name",
+          "type": "string"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        },
+        "properties": {
+          "$ref": "#/definitions/SaasCreationProperties",
+          "description": "Properties of the SaaS resource that are relevant for creation."
+        }
+      }
+    },
+    "SaasProperties": {
```

Comment:
>SaasProperties [](start = 5, length = 14)

Are any of the properties read-only? If yes, they should be marked as such in the swagger.